libraries/chain/db_maint.cpp

```diff
@@ -1079,6 +1079,25 @@ void process_hf_868_890( database& db, bool skip_check_call_orders )
    } // for each market issued asset
 }
 
+/****
+ * @brief A one-time data process to assign the initial taker fee percentage for BSIP81
+ * Every asset issuer may subsequently change the taker fee
+ */
+void process_hf_bsip81(database &db) {
+   // Iterate through every each to set the default taker fee to equal the maker fee
+   // Without this processing code the taker fee will default to zero
+   const auto &asset_idx = db.get_index_type<asset_index>().indices().get<by_type>();
+   for (auto asset_itr = asset_idx.cbegin(); asset_itr != asset_idx.cend(); ++asset_itr) {
+      const asset_object &asset = *asset_itr;
+      const uint16_t raw_market_fee_percent = asset.options.market_fee_percent;
+      wlog("Setting default taker fee of ${asset} to ${raw_market_fee_percent}.",
+              ("asset", asset.symbol) ("raw_market_fee_percent", raw_market_fee_percent));
+      db.modify<asset_object>(asset, [raw_market_fee_percent](asset_object &obj) {
+         obj.options.extensions.value.taker_fee_percent = raw_market_fee_percent;
+      });
+   }
+}
+
```

Comment:
Sorry, this will be no longer needed if we decided to merge https://github.com/bitshares/bsips/pull/271.